internal/handlers/common/error.go

```diff
@@ -108,11 +108,19 @@ func (e *Error) Document() *types.Document {
 	return d
 }
 
+// ProtoErr represents protocol error type.
```

Comment:
It is not clear how that is different from `Error` type.

Reply:
In fact, `Error` type should be `CommandError`. 
Whereas `ProtoErr` represents a common protocol error type (it could be CommandError or WriteErrors).